src/test/groovy/com/jvm_bloggers/core/data_fetching/blogs/json_data/JsonBlogsDataFileValidationSpec.groovy

```diff
@@ -1,59 +1,51 @@
 package com.jvm_bloggers.core.data_fetching.blogs.json_data
 
 import com.google.common.base.CharMatcher
-import com.jvm_bloggers.JvmBloggersApplication
 import groovy.json.JsonSlurper
-import org.springframework.beans.factory.annotation.Value
-import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.core.io.Resource
-import org.springframework.test.context.ActiveProfiles
-import org.springframework.test.context.ContextConfiguration
+import spock.lang.Shared
 import spock.lang.Specification
 import spock.lang.Subject
+import spock.lang.Unroll
 
-@ContextConfiguration
-@SpringBootTest(classes = JvmBloggersApplication)
-@ActiveProfiles("test")
 class JsonBlogsDataFileValidationSpec extends Specification {
 
-    @Value("classpath:blogs/bloggers.json")
-    Resource bloggersJson
+    @Subject
+    @Shared
+    List<BloggerEntry> jsonEntries
 
-    @Value("classpath:blogs/companies.json")
-    Resource companiesJson
+    private static final int TWITTER_USER_TAG_MAXIMUM_LENGTH = 16
 
-    @Value("classpath:blogs/videos.json")
-    Resource videosJson
+    private static final CharMatcher alphanumericOrUnderscore = (CharMatcher.inRange('a' as char, 'z' as char) as CharMatcher)
+            .or(CharMatcher.inRange('A' as char, 'Z' as char) as CharMatcher)
+            .or(CharMatcher.inRange('0' as char, '9' as char) as CharMatcher)
```

Comment:
Guava contains some predefined checks like `JAVA_LETTER_OR_DIGIT`, using some of them should simplify this code a bit :)

Reply:
Cool, I'll check it!

Reply:
@tdziurko I saw them but decided to not use them as they are marked as deprecated(both constants like `JAVA_LETTER_OR_DIGIT` and methods `javaLetterOrDigit()`.
JAVA_LETTER_OR_DIGIT make pass non ascii letters and digits like 'ą' and non Arabic digits like '൯'.
I could use them together `javaLetterOrDigit()` and `ascii()`, so the intersection would be what we want, but I prefer it the way it is (with `javaLetterOrDigit()` beeing deprecated.

Reply:
Sounds good :)